models/repo.go

```diff
@@ -766,7 +766,6 @@ func DeleteRepository(doer *user_model.User, uid, repoID int64) error {
 		&Comment{RefRepoID: repoID},
 		&CommitStatus{RepoID: repoID},
 		&DeletedBranch{RepoID: repoID},
-		&webhook.HookTask{RepoID: repoID},
```

Comment:
But a HookTask belongs to this repository which produced because of an org webhook should also be deleted? Otherwise we will have dirty records?

Reply:
A `HookTask` belongs to a `Webhook`. If a system or organization webhook generates a `HookTask` I would be surprised if this history entry of the webhook gets deleted just because the repository was deleted. The entry should only be deleted if the system/org webhook gets deleted.

Reply:
The hook_task should belong to webhook but also repository. That means, deleting webhook will delete all webhook generated hook tasks. Deleting repository will delete all hook tasks for the repository.

For an individual repository, a webhook will generate hooktask for only one repository. So there is no problem whatever delete repository or delete webhook, we could delete all hooktask according webhook id. But for organization repository, the webhook maybe created and stay a long time(maybe forever? :)). But some repositories created and some repositories deleted. (btw: a possible bug maybe when create a new repository, the system webhook and organization webhook didn't apply into them, I have to check.) So maybe you have never chance to delete all these hook tasks.

Reply:
> The hook_task should belong to webhook but also repository. That means, deleting webhook will delete all webhook generated hook tasks. Deleting repository will delete all hook tasks for the repository.

Same is true for an issue comment but you do not delete the comment just because you delete the user. The comment is related to the issue like the hooktask to the webhook. The history entry of a webhook should not get deleted just because an unused reference is deleted. You don't the system notices of a repository just because the repository isn't there anymore.

> But for organization repository, the webhook maybe created and stay a long time(maybe forever? :)).

Webhooks could stay a long time. The hook tasks of a webhook live up to 7 days. There is a cronjob which deletes them.